### ISSUE TITLE
Dask Parallelization not resource efficient when the aquisition function evalutions are the bottleneck 

### ISSUE THREAD

**Issue by FelixNeutatz** [automl/SMAC3#774]
Dear all,

I wrote a CustomRandomForest class that augments the hyperparameters with additional metadata features. The computation of the metadata features requires some computation. Therefore, the evaluation of the acquisition function is more expensive than usual. When I parallelize it using the Dask parallelization strategy, the resources are not fully used:
![Selection_344](https://user-images.githubusercontent.com/5217389/133761125-77c9a495-8042-4b37-b99e-e85ee5aaf5d1.png)

I suspect that the evaluations of the acquisition function are not parallelized?

However, if I use the pSMAC parallelization approach, it can utilize all resources:
![Selection_345](https://user-images.githubusercontent.com/5217389/133761483-7f7ff6e1-2891-452d-97bf-48264ed2a406.png)

Best regards,
Felix

**Comment by dengdifan** [comment]
Hi,
sorry for the late reply, optimizing the acquisition function values are not parallelized:
https://github.com/automl/SMAC3/blob/master/smac/optimizer/ei_optimization.py#L351

DASK is only applied to run your target function parallelly , but not the acquisition optimization part.

PSMAC applied different strategy to do parallel optimization (it creates several SMAC optimizer in parallel and thus will make full use of resources)

**Comment by TianningGao** [comment]
> Hi, sorry for the late reply, optimizing the acquisition function values are not parallelized: https://github.com/automl/SMAC3/blob/master/smac/optimizer/ei_optimization.py#L351
> 
> DASK is only applied to run your target function parallelly , but not the acquisition optimization part.
> 
> PSMAC applied different strategy to do parallel optimization (it creates several SMAC optimizer in parallel and thus will make full use of resources)

When you say "run target function parallelly", does that mean running the target function observation with one config or multiple configs?

**Comment by dengdifan** [comment]
> When you say "run target function parallelly", does that mean running the target function observation with one config or multiple configs?

With multiple configs. However, the current EPM chooser implementation is not designed for batched optimization, the suggested configurations might gather at a very small region.

**Comment by benjamc** [comment]
Created a new issue to keep track of this potential enhancement.
